synapse/storage/databases/main/event_federation.py

```diff
@@ -73,13 +73,20 @@
 
 logger = logging.getLogger(__name__)
 
-BACKFILL_EVENT_BACKOFF_UPPER_BOUND_SECONDS: int = int(
-    datetime.timedelta(days=7).total_seconds()
-)
+BACKFILL_EVENT_EXPONENTIAL_BACKOFF_MAXIMUM_DOUBLING_STEPS = 8
```

Comment:
This could probably use a comment of what it is / why it is 8.

Reply:
There's a comment below which was aiming at this. (It was the closest I could get to the previous behaviour of 7 days).

Reply:
Right, I found that comment a bit confusing. Will try to make a suggestion.

Reply:
Can the comment go above or does `black` autoformatting not like it?

Reply:
I think this could be simplified a bit to:

```suggestion
# The maximum number of exponential attempts to apply.
#
# A maximum is applied to the number of attempts (instead
# of the overall backoff) to avoid overflowing a 32-bit integer
# while calculating the total number of seconds to backoff.
MAXIMUM_BACKOFF_ATTEMPTS = 8
```

And the full explanation below deleted (and left to either someone coming back into GitHub or put in the commit message itself).

Reply:
> Can the comment go above or does `black` autoformatting not like it?

It can go above, and I think that would help!

Reply:
My understanding is that we don't limit the number of attempts: we'll keep retrying to backfill a 9th, 10th, 11th, ... time etc. We just won't keep doubling the backoff _period_ beyond a limit.

I'll have a think about how best to word things.

> Can the comment go above or does black autoformatting not like it?

It can go above, it's just how I wrote out the logic.

Reply:
> My understanding is that we don't limit the number of attempts: we'll keep retrying to backfill a 9th, 10th, 11th, ... time etc. We just won't keep doubling the backoff _period_ beyond a limit.

Yep, also my understanding. Maybe I worded it poorly. Maybe "limiting" is incorrect and it should be "capping" or something?

Reply:
I wasn't so much limiting versus capping as much as _what_ we're limiting. We're limiting how long the gap between retries is, but we make no limit on the number of retries we'll attempt.

How do you find a91d8101d and 6314b16e2?